vespa-osgi-testrunner/src/main/java/com/yahoo/vespa/testrunner/TestReport.java

```diff
@@ -353,12 +356,18 @@ static void trimStackTraces(Throwable thrown, String testFrameworkRootClass) {
                 break;
             }
         }
+        if (cutoff == 0) {
+            while (--i >= 0 && (   stack[i].isNativeMethod()
+                                || stack[i].getClassName().startsWith("java.lang.")
+                                || stack[i].getClassName().startsWith("java.util.")));
+            cutoff = i + 1;
```

Comment:
JUnit seems to now do a stack trim by default, but a really poor one, leaving a tail of stack frames that we don't want, but which also doesn't contain the info we expected in _our_ trim code, so that also breaks 😢 
This is a best-effort attempt to remedy that. 
But, hmm ... maybe `java.lang.Thread` should be exempt here. That's actually a valid bottom.

Reply:
🤞